common/src/main/java/com/revenuecat/purchases/common/CustomerInfoFactory.kt

```diff
@@ -82,4 +86,44 @@ object CustomerInfoFactory {
             originalPurchaseDate = originalPurchaseDate
         )
     }
+
+    fun buildCustomerInfo(
+        appUserId: String,
+        activePurchasesByHashedToken: Map<String, StoreTransaction>,
+        entitlements: Map<String, String>,
+        products: Set<StoreProduct>
+    ): CustomerInfo {
+        // TODO: should we combine what we have in the cache with this???
+        val subscriptions: Set<StoreTransaction> = activePurchasesByHashedToken.filterValues { it.type == ProductType.SUBS }.values.toSet()
+        val nonSubscriptions = activePurchasesByHashedToken.filterValues { it.type == ProductType.INAPP }.values.toSet()
+        val entitlementInfos = entitlements.buildEntitlementInfos(
+            activePurchasesByHashedToken.values.toSet(),
+            products
+        )
+
+        val expirationDatesByProduct = subscriptions.mapNotNull { subscription ->
+            products.firstOrNull { it.sku == subscription.skus[0] }?.let { product ->
+                    // TODO: should we just compute it to be valid for 24 hours?
```

Comment:
That's what Miguel suggested so that's what I did.

Reply:
Yes I made some changes today to this and I'm just checking if it's an active subscription and setting the expiration to 24 h ahead. I need to clean this up